pkg/scaffold/internal/templates/v2/main.go

```diff
@@ -220,8 +220,11 @@ func main() {
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:             scheme,
 		MetricsBindAddress: metricsAddr,
-		LeaderElection:     enableLeaderElection,
 		Port:               9443, 
+		LeaderElection:     enableLeaderElection,
+		// TODO(user): You MUST ensure LeaderElectionID is unique for your controller. Otherwise, 2 controllers using
+		// the same default LeaderElectionID in the same namespace will race and one of them will stop working silently. 
+		LeaderElectionID: "replace-this-with-a-unique-id",
```

Comment:
I would create a unique string from `domain` and `repo`, adding some hashing too. Something like `"{{ .Domain }}/{{ hash .Repo }}"`.

WDYT?

Reply:
I guess I need to change `file.Template` a bit to make it more flexible to pass the additional `FuncMap` around.
```
type Template interface {
	Builder
	// GetBody returns the template body
	GetBody() string
	// SetTemplateDefaults sets the default values for templates
	SetTemplateDefaults() error
	// GetFuncMap allow each template to provide additional functions which will
	// be merged with the default functions.
	GetFuncMap() template.FuncMap
}
```

Reply:
In that case, I would go for:
```Go
type CustomTemplate interface {
	Template
	GetFuncMap() template.FuncMap
}
```
This way only those that need it have to implement it.

But we could also implement it by having a `HashedRepo` field, set it in the `SetTemplateDefaults`, and then use it like `"{{ .Domain }}/{{ .HashedRepo }}"`.

Reply:
@Adirio Any thoughts on current implementation?

Reply:
The only NIT would be adding the `Template` to the `UseCustomFuncMap`